web/web_test.go

```diff
@@ -153,4 +153,104 @@ func TestMain(m *testing.M) {
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Mattermost Classic App passed browser compatibility.")
+	} else {
+		t.Error("Fail: Mattermost Classic App User Agent Test failed!")
+	}
+
+	//Mattermost App 3.7.1
+	ua = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Mattermost/3.7.1 Chrome/56.0.2924.87 Electron/1.6.11 Safari/537.36"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Mattermost App passed browser compatibility.")
+	} else {
+		t.Error("Fail: Mattermost App User Agent Test failed!")
+	}
+
+	//Franz 4.0.4
+	ua = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Franz/4.0.4 Chrome/52.0.2743.82 Electron/1.3.1 Safari/537.36"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Franz App passed browser compatibility.")
+	} else {
+		t.Error("Fail: Franz App User Agent Test failed!")
+	}
+
+	//Edge 14
+	ua = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Edge User Agent passed browser incompatibility!")
+	} else {
+		t.Error("Fail: Edge should have passed browser compatibility")
+	}
+
+	//IE 11
+	ua = "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: IE 11 passed browser compatibility.")
+	} else {
+		t.Error("Fail: IE 11 User Agent Test failed!")
+	}
+
+	//IE 9
+	ua = "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 7.1; Trident/5.0)"
+	if result := CheckClientCompatability(ua); result == false {
+		t.Log("Pass: IE 9 correctly failed browser compatibility.")
+	} else {
+		t.Error("Fail: IE 9 incorrectly passed browser compatibility!")
+	}
+
+	//Safari 9
+	ua = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Safari/604.1.38"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Safari 9 passed browser compatibility.")
+	} else {
+		t.Error("Fail: Safari 9 User Agent Test failed!")
+	}
+
+	//Safari 8
+	ua = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12"
+	if result := CheckClientCompatability(ua); result == false {
+		t.Log("Pass: Safari 8 correctly failed browser compatibility.")
+	} else {
+		t.Error("Fail: Safari 8 incorrectly passed browser compatibility!")
+	}
+
+	//Safari Mobile
+	ua = "Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B137 Safari/601.1"
+	if result := CheckClientCompatability(ua); result == true {
+		t.Log("Pass: Safari Mobile passed browser compatibility.")
+	} else {
+		t.Error("Fail: Safari Mobile User Agent Test failed!")
+	}
```

Comment:
Some table-driven-testing would make this a lot nicer:

```go
func TestCheckClientCompatability(t *testing.T) {
	for name, tc := range map[string]struct {
		UserAgent string
		Expected  bool
	}{
		"Firefox 40.1": {
			UserAgent: "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.1",
			Expected:  true,
		},
		"Chrome 60": {
			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36",
			Expected:  true,
		},
		"Chrome Mobile": {
			UserAgent: "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Mobile Safari/537.36",
			Expected:  true,
		},
		"Classic app": {
			UserAgent: "Mozilla/5.0 (Linux; Android 8.0.0; Nexus 5X Build/OPR6.170623.013; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/61.0.3163.81 Mobile Safari/537.36 Web-Atoms-Mobile-WebView",
			Expected:  true,
		},
		"Mattermost App 3.7.1": {
			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Mattermost/3.7.1 Chrome/56.0.2924.87 Electron/1.6.11 Safari/537.36",
			Expected:  true,
		},
		"Franz 4.0.4": {
			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Franz/4.0.4 Chrome/52.0.2743.82 Electron/1.3.1 Safari/537.36",
			Expected:  true,
		},
		"Edge 14": {
			UserAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393",
			Expected:  true,
		},
		"IE 11": {
			UserAgent: "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko",
			Expected:  true,
		},
		"IE 9": {
			UserAgent: "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 7.1; Trident/5.0)",
			Expected:  false,
		},
		"Safari 9": {
			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Safari/604.1.38",
			Expected:  true,
		},
		"Safari 8": {
			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12",
			Expected:  false,
		},
		"Safari Mobile": {
			UserAgent: "Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B137 Safari/601.1",
			Expected:  true,
		},
	} {
		t.Run(name, func(t *testing.T) {
			assert.Equal(t, tc.Expected, CheckClientCompatability(tc.UserAgent))
		})
	}
}
```

Reply:
Done. Good Idea!